---
title: "Trait-based approach to bacterial growth efficiency"
author: "Mario E. Muscarella, Xia Meng Howe, Jay T. Lennon"
date: "`r format(Sys.time(), '%d %B, %Y')`"
header-includes:
   - \usepackage{array}
   - \usepackage{graphics}
output: 
  pdf_document:
    fig_caption: true
---

# Initial Setup
```{r results='hide', message=FALSE}
rm(list=ls()) 
setwd("~/GitHub/MicrobialCarbonTraits/analyses")

# Custom Functions
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
sem <- function(x){sd(na.omit(x))/sqrt(length(na.omit(x)))}
CV <- function(x, ...){(sd(x, na.rm = TRUE)/mean(x, na.rm = TRUE))*100}

# Confidence Hulls
add.hull <- function(model = "", pred.frame = ""){
  CI.U <- predict(model, interval = "c", newdata=pred.frame)[, "upr"]
  CI.L <- predict(model, interval = "c", newdata=pred.frame)[, "lwr"]
  pred.frame2 <- unlist(pred.frame)
  X.Vec <- c(pred.frame2, tail(pred.frame2, 1), rev(pred.frame2),
               head(pred.frame2, 1))
  Y.Vec <- c(CI.U, tail(CI.L, 1), rev(CI.L), head(CI.U,1))
  polygon(X.Vec, Y.Vec, col = "gray90", border = NA)
}

levins <- function(p_xi = ""){
  p = 0
  for (i in p_xi){
    p = p + i^2
    }
  nb = 1 / (length(p_xi) * p)
  return(nb)
}

# Save Default Plot Settings
opar <- par(no.readonly = TRUE)  # Saves plot defaults

# Required Packages
#require("muscle");require("seqinr");require("caper")
require("ape");require("phylobase");require("adephylo")
require("geiger");require("picante")
require("stats");require("RColorBrewer")
require("png"); require("grid");require("reshape"); require("car")
```

# Isolate Information
## Figure 1: Phylogenetic Tree
Do I want to make a really nice tree or just use the species tree in the next figure
```{r, results = "hide"}
# Import Phylogenetic Tree
tree.raw <- read.tree("../data/Phylogeny/RAxML_bipartitions.RAxML_HMWF")
tree.rooted <- ape::root(tree.raw, "Aquifex", resolve.root = T)

tree.rooted$node.label

plot.phylo(tree.rooted, type = "phylogram", 
           direction = "right", show.tip.label=TRUE,
           use.edge.length = FALSE, cex = 0.6, label.offset = 1, 
           main = "Seq w/o Neighbors w/o Rooting")
add.scale.bar(cex = 0.7)
nodelabels(tree.rooted$node.label, font = 2, bg = "white", 
           frame = "none", cex = 0.4, adj = c(1.2, 1.2), col = "red")
```

# Bacterial Growth Efficiency
```{r}
# Import Data
BP <- read.csv(file="../data/BP_data.txt") # Produced by CarbonTraitsProduction.R
BR <- read.csv(file="../data/BR_data.txt") # Produced by CarbonTraitsRespiration.R

BP$Organism == BR$Organism

# Remove 15, 17, 32 ()
BP <- BP[-c(11, 13, 21), ]
BR <- BR[-c(11, 13, 21), ]

BGE <- as.data.frame(matrix(NA, 20, 4))
colnames(BGE) <- c("Organism", "BGE_Glu", "BGE_Suc", "BGE_Pro")
BGE$Organism <- BP$Organism

BGE$BGE_Glu <- round(BP$Glu_mean / (BP$Glu_mean + BR$Glu_mean), 5)
BGE$BGE_Suc <- round(BP$Suc_mean / (BP$Suc_mean + BR$Suc_mean), 5)
BGE$BGE_Pro <- round(BP$Pro_mean / (BP$Pro_mean + BR$Pro_mean), 5)

BGE[, 4][which(BGE[, 4] < 0)] <- 0
BGE[, 4][which(BGE[, 4] == "NA")] <- 0
is.na(BGE[, 4]) <- 0

# Save BGE Data
write.csv(BGE, file="../data/BGE_data.txt", quote=FALSE, row.names=FALSE)
```

# Import Tree
```{r}
# Import Phylogeny (already rooted)
tree.in <- read.tree("../data/Phylogeny/RAxML_bipartitions.RAxML_HMWF")

# Need to remove all of the extra seqs. and check bootstraps. 



# Remove Neighbors
Ctrait.samps <- c(as.character(BGE$Organism), "Aquifex")
neighbors <- setdiff(tree.in$tip.label, Ctrait.samps)
tree.samps <- drop.tip(tree.in, neighbors)

# Keep Rooted but Drop Outgroup Branch
tree <- drop.tip(tree.samps, "Aquifex")

#tree$edge.length <- tree$edge.length + 10^-7
```

```{r}
# Carbon Respiration Traits {adephylo}
Resp <- BR[,c(11, 13, 15)]
Resp[is.na(Resp)] <- 0
Resp[Resp > 30] <- 30
colnames(Resp) <- c("Glucose", "Succinate", "Protocatechuate")
rownames(Resp) <- BR$Organism

# Reorder Resp
Resp <- as.matrix(Resp[match(tree$tip.label, row.names(Resp)), ])

# Carbon Production Traits
Prod <- BP[,c(11, 13, 15)]
Prod[is.na(Prod)] <- 0
Prod[Prod < 0] <- 0
Prod[Prod > 10] <- 10
colnames(Prod) <- c("Glucose", "Succinate", "Protocatechuate")
rownames(Prod) <- BP$Organism

# Reorder Prod
Prod <- as.matrix(Prod[match(tree$tip.label, row.names(Prod)), ])

# Carbon BGE Traits
BGE2 <- BGE[,2:4]
BGE2[is.na(BGE2)] <- 0
BGE2[BGE2 < 0] <- 0
BGE2[BGE2 > 0.9] <- 0
colnames(BGE2) <- c("Glu.", "Suc.", "Pro.")
rownames(BGE2) <- BGE$Organism

# Reorder BGE
BGE2 <- as.matrix(BGE2[match(tree$tip.label, row.names(BGE2)), ])
```

##Figure 2: Bacterial Growth Efficiency
```{r, results='hide'}
png(filename="../figures/Figure2.png",
    width = 1200, height = 1200, res = 96*2, bg = "white")
par(opar)

# Subset Tree
tree <- keep.tip(tree.rooted, row.names(BGE2))

# Define Color Palette
mypalette <- colorRampPalette(brewer.pal(9, "YlOrRd"), bias = 2)

# Make Plot
par(mar=c(1,1,1,7) + 0.1)
x <- phylo4d(tree, tip.data = BGE2[], check.node.lables = "as.data")
table.phylo4d(x, treetype = "phylo", symbol = "colors", show.node = FALSE,
              cex.label = 0.8, scale = FALSE, use.edge.length = FALSE,
              edge.color = "black", edge.width = 3, box = FALSE,
              col=mypalette(25), pch = 15, cex.symbol = 2.5,
              ratio.tree = 0.7, cex.legend = 1.25, center = FALSE,
              show.tip.label = FALSE)
mtext(tipLabels(x), side =4, las = 1, at = c(1:20), cex = 0.8)
text(15.5, 19.5, expression(alpha), cex=1.5)
text(15.5, 16.5, expression(beta), cex=1.5)
text(4.5, 10.5, expression(gamma), cex=1.5)
text(11, 13.5, "Xan.", cex=1)
text(11, 10.5, "Aero.", cex=1)
text(11, 6, "Pseu.", cex=1)
text(6, 2, "BGE", cex = 1.5)


dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
# ```

# ```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="BGE"}
img <- readPNG("../figures/Figure2.png")
grid.raster(img)
```

## Statistical Tests
```{r}
# Hypothesis Testing
traits <- cbind(Resp, Prod, BGE2)

# Import Phylogeny (already rooted)
tree1000 <- read.tree("../data/Phylogeny/RAxML_1000bootstrap.HMWF.tre")

# Reorder Traits
tree1000[[1]] <- drop.tip(tree1000[[1]], "Aquifex")
traits <- as.matrix(traits[match(tree1000[[1]]$tip.label, row.names(traits)), ])

# Blomberg's K - BGE
bloms3 <- matrix(NA, 1000, 6)
colnames(bloms3) <- c("K_g", "K_s", "K_p", "P_g", "P_s", "P_p")
# for (i in 1:3){
#   for (j in 1:1000){
#     tree1000[[j]] <- drop.tip(tree1000[[j]], "Aquifex")
#     tree1000[[j]]$edge.length <- tree1000[[j]]$edge.length + 10^-7
#     output <- phylosignal(traits[, 6 + i], tree1000[[j]])
#     bloms3[j, i] <- output[[1]]
#     bloms3[j, 3+i] <- output[[4]]
#   }}
# 
# avgK1 <- colMeans(bloms3)[1:3]
# avgP <- colMeans(bloms3)[4:6]
# length(which(bloms3[, 4] > 0.05)) / 1000
# length(which(bloms3[, 5] > 0.05)) / 1000
# length(which(bloms3[, 5] > 0.05)) / 1000
# 
# 
phylosignal(traits[, 7], tree)
phylosignal(traits[, 8], tree)
phylosignal(traits[, 9], tree)
Kcalc(traits[, 7], tree)
Kcalc(traits[, 8], tree)
Kcalc(traits[, 9], tree)
# 

# Varaince Partitioning
BGE3 <- as.data.frame(BGE2)
BGE3$ID <- row.names(BGE3)
BGE3$Taxon <- c("Pseudo", "Pseudo", "Pseudo", "Pseudo", "Pseudo", "Pseudo",
                "Pseudo", "Pseudo", "Aero", "Aero", "Aero", "Xanto",
                "Xanto", "Xanto", "Beta", "Beta", "Beta", "Alpha", "Alpha", "Alpha")
BGE3$TaxonP <- c("Gamma", "Gamma", "Gamma", "Gamma", 
                "Gamma", "Gamma", "Gamma", "Gamma", "Gamma", "Gamma", "Gamma", "Gamma",
                "Gamma", "Gamma", "Beta", "Beta", "Beta", "Alpha", "Alpha", "Alpha")

# Full Mixed Model
BGE.l <- melt(BGE3)
model <- glm(log1p(BGE.l$value) ~ as.factor(BGE.l$ID))
summary(model)
model <- glm(log1p(BGE.l$value) ~  as.factor(BGE.l$variable))
summary(model)
model <- glm(log1p(BGE.l$value) ~ as.factor(BGE.l$ID) + as.factor(BGE.l$variable))
Anova(model, type = "III")
1 - (0.12178 / 0.40510)
1 - (0.3784 / 0.40510)

library(lme4)
library(MuMIn)
BGE.l <- melt(BGE3)

fm1 <- lme(value ~ as.factor(variable), random=~1|ID, data = BGE.l)
summary(fm1)
r.squaredGLMM(fm1)


fm2 <- lme(value ~ as.factor(Taxon), random=~1|variable, data = BGE.l)
summary(fm2)
r.squaredGLMM(fm2)

fm2 <- lme(value ~ as.factor(ID), random=~1|variable, data = BGE.l)
summary(fm2)
r.squaredGLMM(fm2)


# Mixed Model
BGE.l <- melt(BGE3)
model <- glm(log1p(BGE.l$value) ~ as.factor(BGE.l$Taxon) + as.factor(BGE.l$variable))
summary(model)

1 - (0.29902/0.40510)
1 - (0.35154/0.40510)

Anova(model, type="III")


# Mixed Model
BGE.l <- melt(BGE3)
model <- glm(log1p(BGE.l$value) ~ as.factor(BGE.l$Taxon))
summary(model)

1 - (0.32573/0.40510)

# Mixed Model
BGE.l <- melt(BGE3)
model <- glm(log1p(BGE.l$value) ~ as.factor(BGE.l$TaxonP))
summary(model)

1 - (0.37824/0.40510)

Anova(model, type="III")


# Mixed Model
BGE.l <- melt(BGE3)
model <- glm(log1p(BGE.l$value) ~ as.factor(BGE.l$variable))
summary(model)
1 - (0.3784/0.40510)

Anova(model, type="III")





# Generate Test Statistics for Comparing Phylogenetic Signal {geiger}
#fitContinuous(nj.rooted, nb, model = "lambda")
#fitContinuous(nj.lambda.0, nb, model = "lambda")



# First, Correct for Zero Branch-Lengths on Our Tree
#nj.rooted$edge.length <- nj.rooted$edge.length + 10^-7

# Calculate Phylogenetic Signal for Growth on All Phosphorus Resources
# First, Create a Blank Output Matrix
#p.phylosignal <- matrix(NA, 6, 18)
#colnames(p.phylosignal) <- colnames(p.growth.std)
#rownames(p.phylosignal) <- c("K", "PIC.var.obs", "PIC.var.mean",
#                             "PIC.var.P", "PIC.var.z", "PIC.P.BH")

# Use a For Loop to Calculate Blomberg's K for Each Resource
#for (i in 1:18){
#  x <- as.matrix(p.growth.std[ ,i, drop = FALSE])
#  out <- phylosignal(x, nj.rooted)
#  p.phylosignal[1:5, i] <- round(t(out), 3)
#}

# Use the BH Correction on P-values:
#p.phylosignal[6, ] <- round(p.adjust(p.phylosignal[4, ], method = "BH"), 3)


# Calcualate Phylogenetic Signal for Niche Breadth
#signal.nb <- phylosignal(nb, nj.rooted)

```

# BGE Distribution
```{r, results = "hide"}
png(filename="../figures/Density.png",
    width = 1200, height = 800, res = 96*2, bg = "white")
par(opar)

dip.test(na.omit(BGE2[,1]), simulate.p.value = T)

par(mar = c(4, 4, 1, 1) + 0.5)
plot(density(na.omit(BGE2[,1])), main = "",
     ylab = "", cex.lab = 1.25, 
     cex = 1.5, las = 1, lwd = 2, cex.axis = 1.25)
axis(1, lwd = 2, labels = F, las = 1, cex = 1.25)
axis(2, lwd = 2, labels = F, las = 1, cex = 1.25)
axis(1, lwd = 2, labels = F, tck = 0.02, las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, las = 1)
axis(3, lwd = 2, labels = F, tck = -0.02, las = 1)
axis(4, lwd = 2, labels = F, tck = -0.02, las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, las = 1)
mtext("Density", side = 2, line = 2.5, cex = 1.5)
box(lwd = 2)


dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
#```

#```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Kernel Density Plot"}
img <- readPNG("../figures/Density.png")
grid.raster(img)
```


# Phenotypic Comparisions
```{r}
# Import Data
umax <- read.csv("../data/umax.txt", header=T)
copies <- read.delim("../data/16SrRNA.txt")
ecoplate <- read.delim("../data/eco.data.txt") 
maple.raw <- read.delim("../data/maple.txt", header = T, row.names = 1)

BGE4 <- as.data.frame(BGE2[order(rownames(BGE2)), ])
BGE4$Organism <- rownames(BGE4)
colnames(BGE4) <- c("Glu", "Suc", "Pro", "Organism")

# Raw Data Calculations 
maple <- (maple.raw >= 80) * 1
pathways <- rowSums(maple)
pathways <- data.frame(Organism = names(pathways), pathways)
ecoplate.std <- ecoplate[,2:32] / (apply(ecoplate[,2:32], 1, sum))
ecoplate$levins <- levins(ecoplate.std)

ecoplate2 <- ecoplate[ecoplate$Strain %in% BGE4$Organism, ]
pathways2 <- pathways[pathways$Organism %in% BGE4$Organism, ]

# Build Dataframes
growth <- merge(umax, copies, by.x = "isolate", by.y = "Organism")
growth.bge <- merge(BGE4, growth, by.x = "Organism", by.y = "isolate")

res.niche <- merge(pathways, ecoplate2[,c(1,33,34)], 
               by.x = "Organism", by.y = "Strain", all.x = T)
niche.bge <- merge(BGE4, res.niche)

# Define Groups
growth.bge$group <- NA
for (i in 1:dim(growth.bge)[1])
  if (!is.na(growth.bge$Glu[i])){
if (growth.bge$Glu[i] < 0.05){
  growth.bge$group[i] <- "A"
} else {
  growth.bge$group[i] <- "B"
}} else {}

niche.bge$group <- NA
for (i in 1:dim(niche.bge)[1])
  if (!is.na(niche.bge$Glu[i])){
if (niche.bge$Glu[i] < 0.05){
  niche.bge$group[i] <- "A"
} else {
  niche.bge$group[i] <- "B"
}} else {}

# Physiological Tradeoffs
## Growth Rate
data <- growth.bge
mod1 <- lm(Glu ~ umax * group, data = data)
summary(mod1)

mod2 <- lm(Glu ~ umax, data = data[data$group == "B", ])
summary(mod2)
mod3 <- lm(Glu ~ umax, data = data[data$group == "A", ])
summary(mod3)

## Niche Breadth
data <- niche.bge
mod1 <- lm(Glu ~ levins * group, data = data)
summary(mod1)

mod2 <- lm(Glu ~ levins, data = data[data$group == "B", ])
summary(mod2)
mod3 <- lm(Glu ~ levins, data = data[data$group == "A", ])
summary(mod3)

```

# Figure 3: Species Traits
```{r, results='hide'}
png(filename="../figures/Figure3.png",
    width = 1600, height = 800, res = 96*2, bg = "white")
par(opar)

layout(matrix(1:2, nrow = 1))

par(mar = c(3,0.5,1,0.5) + 0.5, oma = c(2, 5, 1, 1))

# Maximum Growth Rate
groupA <- lm(Glu ~ umax, data = growth.bge[growth.bge$group == "A", ])
groupB <- lm(Glu ~ umax, data = growth.bge[growth.bge$group == "B", ])
pred.frameA <- data.frame(umax = seq(0.04, 0.12, by = 0.01))
pred.frameB <- data.frame(umax = seq(0.05, 0.20, by = 0.01))

plot(Glu ~ umax, growth.bge, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(0.04, 0.20), ylim = c(0,0.35), las = 1,
     pch = 22, bg = "gray", lwd = 2, cex = 1.5)
add.hull(groupA, pred.frameA)
matlines(pred.frameA, predict(groupA, interval = "c", newdata=pred.frameA),
         lty=c(2,3,3), lwd=c(4,2,2), col="black")
add.hull(groupB, pred.frameB)
matlines(pred.frameB, predict(groupB, interval = "c", newdata=pred.frameB),
         lty=c(2,3,3), lwd=c(4,2,2), col="black")
points(Glu ~ umax, growth.bge[growth.bge$group == "A", ],
       pch = 22, bg = "gray", lwd = 2, cex = 1.5)
points(Glu ~ umax, growth.bge[growth.bge$group == "B", ],
       pch = 24, bg = "gray", lwd = 2, cex = 1.5)
axis(1, lwd = 2, labels = T, at = c(seq(0, 0.2, 0.05)), las = 1)
axis(2, lwd = 2, labels = T, at = c(0, 0.1, 0.2, 0.3), las = 1)
axis(1, lwd = 2, labels = F, tck = 0.02, at = c(seq(0, 0.2, 0.05)), las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, at = c(0, 0.1, 0.2, 0.3), las = 1)
axis(3, lwd = 2, labels = F, tck = -0.02, at = c(seq(0, 0.2, 0.05)), las = 1)
axis(4, lwd = 2, labels = F, tck = -0.02, at = c(0, 0.1, 0.2, 0.3), las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, at = c(seq(0, 0.2, 0.05)), las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, at = c(0, 0.1, 0.2, 0.3), las = 1)
mtext(expression(paste("Maximum Growth Rate (hr"^-1,")")), side = 1, line = 3, cex = 1.5)
#mtext("Oligotroph-Copiotroph", side = 3, line = 1)
mtext("Bacterial Growth Efficiency", side = 2, line = 2.5, cex = 1.5)
box(lwd = 2)

# Add Legend
legend("topright", c("High BGE Group", "Low BGE Group"), pch = c(24, 22), pt.bg = "gray", pt.cex = 1, pt.lwd = 2, bty = "n", cex = 0.8)

# Niche Breadth
groupA <- lm(Glu ~ levins, data = niche.bge[niche.bge$group == "A", ])
groupB <- lm(Glu ~ levins, data = niche.bge[niche.bge$group == "B", ])
pred.frameA <- data.frame(levins = seq(0.04, 0.45, by = 0.01))
pred.frameB <- data.frame(levins = seq(0.04, 0.5, by = 0.01))

plot(Glu ~ levins, niche.bge, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(0.04, 0.5), ylim = c(0, 0.35), las = 1,
     pch = 22, bg = "gray", lwd = 2, cex = 1.5)
add.hull(groupA, pred.frameA)
matlines(pred.frameA, predict(groupA, interval = "c", newdata=pred.frameA),
         lty=c(2,3,3), lwd=c(4,2,2), col="black")
add.hull(groupB, pred.frameB)
matlines(pred.frameB, predict(groupB, interval = "c", newdata=pred.frameB),
         lty=c(2,3,3), lwd=c(4,2,2), col="black")
points(Glu ~ levins, niche.bge[niche.bge$group == "A", ],
       pch = 22, bg = "gray", lwd = 2, cex = 1.5)
points(Glu ~ levins, niche.bge[niche.bge$group == "B", ],
       pch = 24, bg = "gray", lwd = 2, cex = 1.5)
axis(1, lwd = 2, labels = T, at = c(seq(0, 0.5, 0.1)), las = 1)
axis(2, lwd = 2, labels = F, at = c(0, 0.1, 0.2, 0.3), las = 1)
axis(1, lwd = 2, labels = F, tck = 0.02, at = c(seq(0, 0.5, 0.1)), las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, at = c(0, 0.1, 0.2, 0.3), las = 1)
axis(3, lwd = 2, labels = F, tck = -0.02, at = c(seq(0, 0.5, 0.1)), las = 1)
axis(4, lwd = 2, labels = F, tck = -0.02, at = c(0, 0.1, 0.2, 0.3), las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, at = c(seq(0, 0.5, 0.1)), las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, at = c(0, 0.1, 0.2, 0.3), las = 1)
mtext("Levin's Index", side = 1, line = 3, cex = 1.5)
#mtext("Generalist-Specialist", side = 3, line = 1)
box(lwd = 2)

dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
#```

#```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Traits"}
img <- readPNG("../figures/Figure3.png")
grid.raster(img)
```



# Genomic Comparisions
```{r}
# Number of Pathways
data <- niche.bge
mod1 <- lm(Glu ~ pathways * group, data = data)
summary(mod1)

mod2 <- lm(Glu ~ pathways, data = data[data$group == "B", ])
summary(mod2)

t.test(data$pathways ~ data$group)

# Permanova
str(maple)
maple2 <- maple[BGE4$Organism, ]
maple2 <- maple2[, which(colSums(maple2) > 0)]

rownames(maple2) == data$Organism

t.test(rowSums(maple2) ~ data$group)

adonis(maple2 ~ as.factor(data$group), method = "bray")

library(labdsv)
inds <- indval(maple2, data$group)
summary(inds)


# Just in group B
mapleB <- maple2[data$group == "B", ]
BGE_glu_B <- BGE4$Glu[data$group == "B"]

adonis(mapleB ~ BGE_glu_B, method = "bray")

# Calculate Bray-Curtis Distances for Genomes
mapleB.bray <- vegdist(mapleB, "bray")

# BGE dbRDA 
BGE.dbRDA.B <- capscale(mapleB.bray ~ BGE_glu_B, comm = mapleB, 
                      add = T, na.action = na.exclude)
summary(BGE.dbRDA.B)
RsquareAdj(BGE.dbRDA.B)
anova(BGE.dbRDA.B)

# Just in group A
mapleA <- maple2[data$group == "A", ]
BGE_glu_A <- BGE4$Glu[data$group == "A"]

adonis(mapleA ~ BGE_glu_A, method = "bray")

# Calculate Bray-Curtis Distances for Genomes
mapleA.bray <- vegdist(mapleA, "bray")

# BGE dbRDA 
BGE.dbRDA.A <- capscale(mapleA.bray ~ BGE_glu_A, comm = mapleA, 
                      add = T, na.action = na.exclude)
summary(BGE.dbRDA.A)
RsquareAdj(BGE.dbRDA.A)
anova(BGE.dbRDA.A)

# Pathways Scores
pathway.scores <- data.frame(matrix(NA, 108, 2))
colnames(pathway.scores) <- c("pathway", "cor")
pathway.scores[,1] <- colnames(mapleA)
for (i in 1:108){
  pathway.scores[i,2] <- cor(BGE_glu_A,mapleA[,i], method = "spearman")
  #pathway.scores[i,2] <- cor(BGE.dbRDA.B$CCA$wa,mapleB[,i], method = "spearman")
}

sig.path <- pathway.scores[which(abs(pathway.scores$cor) >= 0.70), ]
sig.path





# Group Differences
growth.bge
growth.bge$group

groups <- growth.bge[which(!is.na(growth.bge$BGE_Glu)), c(1, 7)]
maple.raw <- read.delim("../data/Maple/maple.txt", header = T, row.names = 1)
maple80 <- (maple.raw >= 80) * 1
maple60 <- (maple.raw >= 60) * 1

maplePA <- maple80[, which(colSums(maple80) >=1)]
maple1 <- maple80[which(rownames(maple80) %in% groups$Organism), ]
maple1 <- maple1[, which(colSums(maple1) >=1)]
maple2 <- maple60[which(rownames(maple60) %in% groups$Organism), ]
maple2 <- maple2[, which(colSums(maple2) >= 1)]
rownames(maple1)
setdiff(as.character(groups$Organism), rownames(maple1))
setdiff(rownames(maple1), as.character(groups$Organism))


t.test(rowSums(maplePA) ~ groups$group, var.equal = T)

groups <- data.frame(Organisms = BGE$Organism[which(!is.na(BGE$BGE_Glu))], group = NA)
for (i in 1:dim(groups)[1])
  if (!is.na(BGE$BGE_Glu[groups$Organisms[i]])){
if (BGE$BGE_Glu[groups$Organisms[i]] < 0.05){
  groups$group[i] <- "A"
} else {
  groups$group[i] <- "B"
}} else {}

adonis(maple1 ~ as.factor(groups$group), method = "bray")

library(labdsv)
inds <- indval(maple1, groups$group)
summary(inds)

# Group A
# M00168 CAM (Crassulacean acid metabolism), dark
# M00169 CAM (Crassulacean acid metabolism), light

# Group B
# M00124 Pyridoxal biosynthesis, erythrose-4P => pyridoxal-5P
# M00045 Histidine degradation, histidine => N-formiminoglutamate => glutamate
# M00565 Trehalose biosynthesis, D-glucose 1P => trehalose


inds <- indval(maple2, groups$group)
summary(inds)

# Just in group B
mapleB <- maple[groups$Organisms[groups$group == "B"], ]
BGE_glu_B <- BGE$BGE_Glu[groups$Organisms[groups$group == "B"]]

adonis(mapleB ~ BGE_glu_B, method = "bray")

# Calculate Bray-Curtis Distances for Genomes
mapleB.bray <- vegdist(mapleB, "bray")

# BGE dbRDA 
BGE.dbRDA.B <- capscale(mapleB.bray ~ BGE_glu_B, comm = mapleB, 
                      add = T, na.action = na.exclude)
summary(BGE.dbRDA.B)
RsquareAdj(BGE.dbRDA.B)
anova(BGE.dbRDA.B)

# DOM Scores
pathway.scores <- data.frame(matrix(NA, 108, 2))
colnames(pathway.scores) <- c("pathway", "cor")
pathway.scores[,1] <- colnames(mapleB)
for (i in 1:108){
  pathway.scores[i,2] <- cor(BGE_glu_B,mapleB[,i], method = "spearman")
  #pathway.scores[i,2] <- cor(BGE.dbRDA.B$CCA$wa,mapleB[,i], method = "spearman")
}

sig.path <- pathway.scores[which(abs(pathway.scores$cor) >= 0.70), ]

# Positive
# M00173 Reductive citrate cycle (Arnon-Buchanan cycle)
# M00579 Phosphate acetyltransferase-acetate kinase pathway, acetyl-CoA => acetate
# M00632 Galactose degradation, Leloir pathway, galactose => alpha-D-glucose-1P
# M00064 ADP-L-glycero-D-manno-heptose biosynthesis
# M00554 Nucleotide sugar biosynthesis, galactose => UDP-galactose
# M00565 Trehalose biosynthesis, D-glucose 1P => trehalose
# M00017 Methionine biosynthesis, apartate => homoserine => methionine
# M00027 GABA (gamma-Aminobutyrate) shunt
# M00117 Ubiquinone biosynthesis, prokaryotes, chorismate => ubiquinone
# M00116 Menaquinone biosynthesis, chorismate => menaquinone
# M00140 C1-unit interconversion, prokaryotes

# Negative
# M00004 Pentose phosphate pathway (Pentose phosphate cycle)
# M00008 Entner-Doudoroff pathway, glucose-6P => glyceraldehyde-3P + pyruvate
# M00552 D-galactonate degradation, De Ley-Doudoroff pathway, D-galactonate => glycerate-3P
# M00119 Pantothenate biosynthesis, valine/L-aspartate => pantothenate
# M00122 Cobalamin biosynthesis, cobinamide => cobalamin












maple.raw <- read.delim("../data/Maple/maple.txt", header = T, row.names = 1)
maple <- (maple.raw >= 80) * 1
maple <- maple[BGE$Organism,which(colSums(maple) > 1)]

maple2 <- maple[-c(which(is.na(BGE$BGE_Glu))),]
maple2 <- maple2[,which(colSums(maple2) > 1)]

# Calculate Bray-Curtis Distances for Genomes
maple.bray <- vegdist(maple, "bray")

# BGE dbRDA 
BGE.dbRDA <- capscale(maple.bray ~ BGE$BGE_Glu, comm = maple, 
                      add = T, na.action = na.exclude)
summary(BGE.dbRDA)
RsquareAdj(BGE.dbRDA)
anova(BGE.dbRDA)
str(BGE.dbRDA$CCA$wa)
str(maple2)


# DOM Scores
pathway.scores <- data.frame(matrix(NA, 94, 2))
colnames(pathway.scores) <- c("pathway", "cor")
pathway.scores[,1] <- colnames(maple2)
for (i in 1:86){
  pathway.scores[i,2] <- cor(BGE.dbRDA$CCA$wa,maple2[,i], method = "spearman")
}

sig.path <- pathway.scores[which(abs(pathway.scores$cor) > 0.70), ]

plot(BGE.dbRDA$CCA$wa, na.omit(BGE$BGE_Glu))

# Neg
# M00169	Pathway	CAM (Crassulacean acid metabolism), light

# Pos
# M00045	Pathway	Histidine degradation, histidine => N-formiminoglutamate => glutamate
# M00124	Pathway	Pyridoxal biosynthesis, erythrose-4P => pyridoxal-5P
# M00133	Pathway	Polyamine biosynthesis, arginine => agmatine => putrescine => spermidine

sum(maple2[,"M00169"] > 0)
plot(maple2[,"M00169"],na.omit(BGE$BGE_Glu))

sum(na.omit(BGE$BGE_Glu) < 0.01)




```









# Physiological Constraints
```{r}
# Organize Data
BGE4 
BGE4$Group <- NA
for (i in 1:dim(BGE4)[1])
  if (!is.na(BGE4$Glu[i])){
if (BGE4$Glu[i] < 0.05){
  BGE4$Group[i] <- "A"
} else {
  BGE4$Group[i] <- "B"
}} else {}


Phys <- data.frame(Organism = BR$Organism,
                   RespGlu = BR$Glu_mean, RespSuc = BR$Suc_mean, 
                   RespPro = BR$Pro_mean, 
                   ProdGlu = BP$Glu_mean, ProdSuc = BP$Suc_mean, 
                   ProdPro = BP$Pro_mean, 
                   BGEGlu = BGE4$Glu, BGESuc = BGE4$Suc,
                   BGEPro = BGE4$Pro,
                   Group = BGE4$Group)

# Full Model
mod1 <- lm(log1p(ProdGlu) ~ log1p(RespGlu) * Group, data = Phys)
summary(mod1)

# Independent Models
modA <- lm(log1p(ProdGlu) ~ log1p(RespGlu), data = Phys[Phys$Group == "A", ])
summary(modA)
modB <- lm(log1p(ProdGlu) ~ log1p(RespGlu), data = Phys[Phys$Group == "B", ])
summary(modB)

Phys2 <- Phys
Phys2$RespGlu <- log10(Phys2$RespGlu)
Phys2$ProdGlu <- log10(Phys2$ProdGlu)
Phys2$RespSuc <- log10(Phys2$RespSuc)
Phys2$ProdSuc <- log10(Phys2$ProdSuc)
Phys2$RespPro <- log10(Phys2$RespPro)
Phys2$ProdPro <- log10(Phys2$ProdPro)
for (i in 2:7){
  Phys2[which(Phys2[, i] < -5), i] = 0 
}


# Maximum Growth Rate
glu.model <- lm(ProdGlu ~ RespGlu * Group, data = Phys2)
suc.model <- lm(ProdSuc ~ RespSuc * Group, data = Phys2)
pro.model <- lm(ProdPro ~ RespPro * Group, data = Phys2)


# Compare Models
PhysGlu <- Phys2[, c(2, 5, 11)]
colnames(PhysGlu) <- c("Resp", "Prod", "Group")
PhysSuc <- Phys2[, c(3, 6, 11)]
colnames(PhysSuc) <- c("Resp", "Prod", "Group")

glu.model <- lm(Prod ~ Resp * Group, data = PhysGlu)
suc.model <- lm(Prod ~ Resp * Group, data = PhysSuc)

test <- anova(glu.model, suc.model, test = "F")

PhysGlu <- Phys2[, c(1, 2, 5, 8, 11)]
colnames(PhysGlu) <- c("Organism", "Resp", "Prod", "BGE", "Group")
PhysGlu$Res <- rep("Glu", dim(PhysGlu)[1])
PhysSuc <- Phys2[, c(1, 3, 6, 9, 11)]
colnames(PhysSuc) <- c("Organism", "Resp", "Prod", "BGE", "Group")
PhysSuc$Res <- rep("Suc", dim(PhysSuc)[1])
PhysPro <- Phys2[, c(1, 4, 7, 10, 11)]
colnames(PhysPro) <- c("Organism", "Resp", "Prod", "BGE", "Group")
PhysPro$Res <- rep("Pro", dim(PhysPro)[1])
Phys.l <- rbind(PhysGlu, PhysSuc, PhysPro)
Phys.l2 <- na.omit(Phys.l)

Phys.l.b <- Phys.l2
Phys.l.b$Resp <- 10*exp(Phys.l.b$Resp)
Phys.l.b$Prod <- 10*exp(Phys.l.b$Prod)
Phys.l.b$BGE <- 10*exp(Phys.l.b$BGE)

mod1 <- lm(Resp ~ (Prod + Res + Group)^2, data = Phys.l2)
summary(mod1)

ttest <- function(reg, coefnum, val){
  co <- coef(summary(reg))
  tstat <- (co[coefnum,1]-val)/co[coefnum,2]
  pstat <- 2 * pt(abs(tstat), reg$df.residual, lower.tail = FALSE)
  return(list = c(tstat, reg$df.residual, pstat))
}

ttest(mod1, 2, 1)



```


```{r, results = "hide"}
png(filename="../figures/Figure5.png",
    width = 1200, height = 1200, res = 96*2, bg = "white")
par(opar)


par(mar = c(6,6,1,1) + 0.5)
groupA1 <- lm(Resp ~ Prod, 
              data = Phys.l2[Phys.l2$Group == "A" & Phys.l2$Res == "Glu", ])
groupB1 <- lm(Resp ~ Prod, 
              data = Phys.l2[Phys.l2$Group == "B" & Phys.l2$Res == "Glu", ])
groupA <- lm(Resp ~ Prod, data = Phys.l2[Phys.l2$Group == "A", ])
groupB <- lm(Resp ~ Prod, data = Phys.l2[Phys.l2$Group == "B", ])
groupA.b <- lm(Resp ~ Prod, data = Phys.l.b[Phys.l.b$Group == "A", ])
groupB.b <- lm(Resp ~ Prod, data = Phys.l.b[Phys.l.b$Group == "B", ])
pred.frameA <- data.frame(Prod = seq(-3.2, 1.5, by = 0.1))
pred.frameB <- data.frame(Prod = seq(-0.5, 2, by = 0.1))
pred.frameC <- data.frame(Prod = seq(-3, -0.5, by = 0.1))
pred.frameD <- data.frame(Prod = seq(-7, -3.2, by = 0.1))

plot(Resp ~ Prod, Phys.l2, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(-4, 2.2), ylim = c(-1,3.4), las = 1,
     pch = 22, bg = "gray", lwd = 2, cex = 1.5)
add.hull(groupA1, pred.frameA)
matlines(pred.frameA, predict(groupA1, interval = "c", newdata=pred.frameA),
        lty=c(2,3,3), lwd=c(4,2,2), col="black") 
add.hull(groupB1, pred.frameB)
matlines(pred.frameB, predict(groupB1, interval = "c", newdata=pred.frameB),
         lty=c(2,3,3), lwd=c(4,2,2), col="black")  

matlines(pred.frameC, predict(groupB1, interval = "c", newdata=pred.frameC),
         lty=c(2,0,0), lwd=c(2,0,0), col="black")
matlines(pred.frameD, predict(groupA1, interval = "c", newdata=pred.frameD),
         lty=c(2,0,0), lwd=c(2,0,0), col="black")

points(Resp ~ Prod, Phys.l2[Phys.l2$Group == "A" & Phys.l2$Res == "Glu", ],
       pch = 22, bg = "gray80", lwd = 2, 
       cex = 1.2 + Phys.l2$BGE[Phys.l2$Group == "A" & Phys.l2$Res == "Glu"] * 5) 
points(Resp ~ Prod, Phys.l2[Phys.l2$Group == "B" & Phys.l2$Res == "Glu", ],
       pch = 24, bg = "gray80", lwd = 2, 
       cex = 1.2 + Phys.l2$BGE[Phys.l2$Group == "B" & Phys.l2$Res == "Glu"] * 5) 

points(Resp ~ Prod, Phys.l2[Phys.l2$Group == "A" & Phys.l2$Res == "Suc", ],
       pch = 22, bg = "gray60", lwd = 2, 
       cex = 1.2 + Phys.l2$BGE[Phys.l2$Group == "A" & Phys.l2$Res == "Suc"] * 5) 
points(Resp ~ Prod, Phys.l2[Phys.l2$Group == "B" & Phys.l2$Res == "Suc", ],
       pch = 24, bg = "gray60", lwd = 2, 
       cex = 1.2 + Phys.l2$BGE[Phys.l2$Group == "B" & Phys.l2$Res == "Suc"] * 5) 

points(Resp ~ Prod, Phys.l2[Phys.l2$Group == "A" & Phys.l2$Res == "Pro", ],
       pch = 22, bg = "gray40", lwd = 2, 
       cex = 1.2 + Phys.l2$BGE[Phys.l2$Group == "A" & Phys.l2$Res == "Pro"] * 5) 
points(Resp ~ Prod, Phys.l2[Phys.l2$Group == "B" & Phys.l2$Res == "Pro", ],
       pch = 24, bg = "gray40", lwd = 2, 
       cex = 1.2 + Phys.l2$BGE[Phys.l2$Group == "B" & Phys.l2$Res == "Pro"] * 5) 

abline(v = 0)
abline(h = 1)
abline(h = 0)

text(-3.8, 0.52, paste("*", round(groupA1$coefficients[1], 2), sep = ""), cex = 1.2)
text(-2, -1, paste("*", round(groupB1$coefficients[1], 2), sep = ""), , cex = 1.2) 

axis(1, lwd = 2, labels = c(0.001, 0.1, 100), at = c(seq(-3, 2, 2)), las = 1)
axis(2, lwd = 2, labels = c(0.01, 0.1, 1, 10, 100, 1000), at = c(seq(-2, 3, 1)), las = 1)
axis(1, lwd = 2, labels = F, tck = 0.02, at = c(seq(-3, 3, 2)), las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, at = c(seq(-2, 3, 1)), las = 1)
axis(3, lwd = 2, labels = F, tck = -0.02, at = c(seq(-3, 3, 2)), las = 1)
axis(4, lwd = 2, labels = F, tck = -0.02, at = c(seq(-2, 3, 1)), las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, at = c(seq(-3, 3, 2)), las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, at = c(seq(-2, 3, 1)), las = 1)
mtext("Respiration Rate", side = 2, line = 4.5, cex = 1.5)
mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
                       side = 2, line = 3, cex = 1)
mtext("Production Rate", side = 1, line = 2.5, cex = 1.5)
mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
                       side = 1, line = 4, cex = 1)
box(lwd = 2)

# Add Legend
legend("bottomright", c("High BGE Group", "Low BGE Group"), 
       pch = c(24, 22), pt.bg = "gray", pt.cex = 1, pt.lwd = 2, 
       bty = "n", cex = 0.8, inset = 0.05)
legend("topleft", c("Glucose", "Succinate", "Protocatechuate"), 
       pch = 22, pt.bg = c("gray80", "gray60", "gray40"), pt.cex = 1, pt.lwd = 2, 
       bty = "n", cex = 0.8, inset = 0.05)


dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
# ```

# ```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="BGE"}
img <- readPNG("../figures/Figure5.png")
grid.raster(img)
```


```{r, results='hide'}
png(filename="../figures/AllResources.png",
    width = 800, height = 1800, res = 96*2, bg = "white")
par(opar)

layout(matrix(1:3, nrow = 3))

par(mar = c(0.25,6,0.25,1) + 0.5, oma = c(7, 1, 1, 1) + 0.5)

# Glucose
groupA <- lm(ProdGlu ~ RespGlu, data = Phys2[Phys2$Group == "A", ])
groupB <- lm(ProdGlu ~ RespGlu, data = Phys2[Phys2$Group == "B", ])
pred.frameA <- data.frame(RespGlu = seq(0.5, 3.2, by = 0.1))
pred.frameB <- data.frame(RespGlu = seq(0.3, 1.8, by = 0.1))

plot(ProdGlu ~ RespGlu, Phys2, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(0.3, 3.2), ylim = c(-3,2.2), las = 1,
     pch = 22, bg = "gray", lwd = 2, cex = 1.5)
add.hull(groupA, pred.frameA)
matlines(pred.frameA, predict(groupA, interval = "c", newdata=pred.frameA),
         lty=c(2,3,3), lwd=c(4,2,2), col="black") 
add.hull(groupB, pred.frameB)
matlines(pred.frameB, predict(groupB, interval = "c", newdata=pred.frameB),
         lty=c(2,3,3), lwd=c(4,2,2), col="black")  
points(ProdGlu ~ RespGlu, Phys2[Phys2$Group == "A", ],
       pch = 22, bg = "gray", lwd = 2, cex = 1.2 + Phys2$BGEGlu[Phys2$Group == "A"] * 5) 
points(ProdGlu ~ RespGlu, Phys2[Phys2$Group == "B", ],
       pch = 24, bg = "gray", lwd = 2, cex = 1.2 + Phys2$BGEGlu[Phys2$Group == "B"] * 5)  
axis(1, lwd = 2, labels = F, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(2, lwd = 2, labels = c(0.001, 0.01, 0.1, 1, 10, 100), at = c(seq(-3, 2, 1)), las = 1)
axis(1, lwd = 2, labels = F, tck = 0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, at = c(seq(-3, 2, 1)), las = 1)
axis(3, lwd = 2, labels = F, tck = -0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(4, lwd = 2, labels = F, tck = -0.02, at = c(seq(-3, 2, 1)), las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, at = c(seq(-3, 2, 1)), las = 1)
#mtext("Respiration Rate", side = 1, line = 2.5, cex = 1.5)
#mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
#                       side = 1, line = 4, cex = 1)
#mtext("Production Rate", side = 2, line = 4.5, cex = 1.5)
#mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
#                       side = 2, line = 3, cex = 1)
box(lwd = 2)

# Add Legend
#legend("bottomright", c("High BGE Group", "Low BGE Group"), 
 #      pch = c(24, 22), pt.bg = "gray", pt.cex = 1, pt.lwd = 2, 
 #      bty = "n", cex = 0.8)
legend("topleft", "Glucose", bty = "n")

# Succinate
groupA <- lm(ProdSuc ~ RespSuc, data = Phys2[Phys2$Group == "A", ])
groupB <- lm(ProdSuc ~ RespSuc, data = Phys2[Phys2$Group == "B", ])
pred.frameA <- data.frame(RespSuc = seq(0.7, 2.5, by = 0.01))
pred.frameB <- data.frame(RespSuc = seq(0.89, 2.77, by = 0.01))

plot(ProdSuc ~ RespSuc, Phys2, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(0.3, 3.2), ylim = c(-3.2, 2.2), las = 1,
     pch = 22, bg = "gray", lwd = 2, cex = 1.5)
add.hull(groupA, pred.frameA)
matlines(pred.frameA, predict(groupA, interval = "c", newdata=pred.frameA),
         lty=c(2,3,3), lwd=c(4,2,2), col="black") 
add.hull(groupB, pred.frameB)
matlines(pred.frameB, predict(groupB, interval = "c", newdata=pred.frameB),
         lty=c(2,3,3), lwd=c(4,2,2), col="black")  
points(ProdSuc ~ RespSuc, Phys2[Phys2$Group == "A", ],
       pch = 22, bg = "gray", lwd = 2, cex = 1.2 + Phys2$BGESuc[Phys2$Group == "A"] * 5) 
points(ProdSuc ~ RespSuc, Phys2[Phys2$Group == "B", ],
       pch = 24, bg = "gray", lwd = 2, cex = 1.2 + Phys2$BGESuc[Phys2$Group == "B"] * 5)  
axis(1, lwd = 2, labels = F, at = c(0.3, seq(1, 4, 1)), las = 1)
axis(2, lwd = 2, labels = c(0.001, 0.01, 0.1, 1, 10, 100), at = c(seq(-3, 2, 1)), las = 1)
axis(1, lwd = 2, labels = F, tck = 0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, at = c(seq(-3, 2, 1)), las = 1)
axis(3, lwd = 2, labels = F, tck = -0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(4, lwd = 2, labels = F, tck = -0.02, at = c(seq(-3, 2, 1)), las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, at = c(seq(-3, 2, 1)), las = 1)
#mtext("Respiration Rate", side = 1, line = 2.5, cex = 1.5)
#mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
#                       side = 1, line = 4, cex = 1)
mtext("Production Rate", side = 2, line = 5, cex = 1.5)
mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
                       side = 2, line = 3.5, cex = 1)
box(lwd = 2)

# Add Legend
#legend("bottomright", c("High BGE Group", "Low BGE Group"), 
#       pch = c(24, 22), pt.bg = "gray", pt.cex = 1, pt.lwd = 2, 
#       bty = "n", cex = 0.8, inset = 0.02)
legend("topleft", "Succinate", bty = "n")

# Protocatechuate
pro.model<- lm(ProdPro ~ RespPro, data = Phys2)
groupA <- lm(ProdPro ~ RespPro, data = Phys2[Phys2$Group == "A", ])
groupB <- lm(ProdPro ~ RespPro, data = Phys2[Phys2$Group == "B", ])
pred.frameA <- data.frame(RespPro = seq(1, 3.2, by = 0.1))
pred.frameB <- data.frame(RespPro = seq(0.89, 2.77, by = 0.01))

plot(ProdPro ~ RespPro, Phys2, axes = F, type = "n", 
     xlab = "", ylab = "",
     xlim = c(0.3, 3.2), ylim = c(-2.2, 2.2), las = 1,
     pch = 22, bg = "gray", lwd = 2, cex = 1.5) 
add.hull(pro.model, pred.frameA)
matlines(pred.frameA, predict(pro.model, interval = "c", newdata=pred.frameA),
         lty=c(2,3,3), lwd=c(4,2,2), col="black") 
#add.hull(groupB, pred.frameB)
#matlines(pred.frameB, predict(groupB, interval = "c", newdata=pred.frameB),
#         lty=c(2,3,3), lwd=c(4,2,2), col="black")  
points(ProdPro ~ RespPro, Phys2[Phys2$Group == "A", ],
       pch = 22, bg = "gray", lwd = 2, cex = 1.2 + Phys2$BGEPro[Phys2$Group == "A"] * 5) 
points(ProdPro ~ RespPro, Phys2[Phys2$Group == "B", ],
       pch = 24, bg = "gray", lwd = 2, cex = 1.2 + Phys2$BGEPro[Phys2$Group == "B"] * 5)  
axis(1, lwd = 2, labels = c(2, 10, 100, 1000, 10000), at = c(0.3, seq(1, 4, 1)), las = 1)
axis(2, lwd = 2, labels = c(0.001, 0.01, 0.1, 1, 10, 100), at = c(seq(-3, 2, 1)), las = 1)
axis(1, lwd = 2, labels = F, tck = 0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, at = c(seq(-3, 2, 1)), las = 1)
axis(3, lwd = 2, labels = F, tck = -0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(4, lwd = 2, labels = F, tck = -0.02, at = c(seq(-3, 2, 1)), las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, at = c(seq(-3, 2, 1)), las = 1)
mtext("Respiration Rate", side = 1, line = 3.5, cex = 1.5)
mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
                       side = 1, line = 5, cex = 1)
#mtext("Production Rate", side = 2, line = 4.5, cex = 1.5)
#mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
#                       side = 2, line = 3, cex = 1)
box(lwd = 2)

# Add Legend
legend("bottomright", c("High BGE Group", "Low BGE Group"), pch = c(24, 22), pt.bg = "gray", pt.cex = 1, pt.lwd = 2, bty = "n", cex = 0.8, inset = 0.02)
legend("topleft", "Protocatechuate", bty = "n")


dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
# ```

# ```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="BGE"}
img <- readPNG("../figures/AllResources.png")
grid.raster(img)
```


