---
title: "Trait-based approach to bacterial growth efficiency"
author: "Mario E. Muscarella, Xia Meng Howe, Jay T. Lennon"
date: "`r format(Sys.time(), '%d %B, %Y')`"
header-includes:
   - \usepackage{array}
   - \usepackage{graphics}
output: 
  pdf_document:
    fig_caption: true
---

# Overview
This manuscript explores bacterial growth efficiency (BGE) using a trait-based approach. We used 20 bacterial isolates and three different carbon substrates to test how consumer and resource identity interact to determine BGE

Note: Fig 1 shows the three different Carbon substrates used in this manuscript.

# Initial Setup
```{r results='hide', message=FALSE}
rm(list=ls()) 
setwd("~/GitHub/MicrobialCarbonTraits/analyses")

# Custom Functions
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
sem <- function(x){sd(na.omit(x))/sqrt(length(na.omit(x)))}
CV <- function(x, ...){(sd(x, na.rm = TRUE)/mean(x, na.rm = TRUE))*100}

# Confidence Hulls
add.hull <- function(model = "", pred.frame = ""){
  CI.U <- predict(model, interval = "c", newdata=pred.frame)[, "upr"]
  CI.L <- predict(model, interval = "c", newdata=pred.frame)[, "lwr"]
  pred.frame2 <- unlist(pred.frame)
  X.Vec <- c(pred.frame2, tail(pred.frame2, 1), rev(pred.frame2),
               head(pred.frame2, 1))
  Y.Vec <- c(CI.U, tail(CI.L, 1), rev(CI.L), head(CI.U,1))
  polygon(X.Vec, Y.Vec, col = "gray90", border = NA)
}

levins <- function(p_xi = ""){
  p = 0
  for (i in p_xi){
    p = p + i^2
    }
  nb = 1 / (length(p_xi) * p)
  return(nb)
}

# Save Default Plot Settings
opar <- par(no.readonly = TRUE)  # Saves plot defaults

# Required Packages
library("ape");library("phylobase");library("adephylo")
library("geiger");library("picante"); library("phytools")
library("stats");library("RColorBrewer"); library("labdsv")
library("png"); library("grid");library("reshape"); library("car")
library("lme4");library("MuMIn"); library("diptest"); library("nlme")

source("../bin/Phylo4D.R")
```


# Bacterial Growth Efficiency
## Import Data
```{r}
# Import Data
BP <- read.csv(file="../data/BP_data.txt") # Produced by CarbonTraitsProduction.R
BR <- read.csv(file="../data/BR_data.txt") # Produced by CarbonTraitsRespiration.R

BP$Organism == BR$Organism

# Remove 15, 17, 32 ()
BP <- BP[-c(11, 13, 21), ]
BR <- BR[-c(11, 13, 21), ]

BGE <- as.data.frame(matrix(NA, 20, 4))
colnames(BGE) <- c("Organism", "BGE_Glu", "BGE_Suc", "BGE_Pro")
BGE$Organism <- BP$Organism

BGE$BGE_Glu <- round(BP$Glu_mean / (BP$Glu_mean + BR$Glu_mean), 5)
BGE$BGE_Suc <- round(BP$Suc_mean / (BP$Suc_mean + BR$Suc_mean), 5)
BGE$BGE_Pro <- round(BP$Pro_mean / (BP$Pro_mean + BR$Pro_mean), 5)

BGE[, 4][which(BGE[, 4] < 0)] <- 0
BGE[, 4][which(BGE[, 4] == "NA")] <- 0
is.na(BGE[, 4]) <- 0

range(na.omit(c(BGE$BGE_Glu, BGE$BGE_Pro, BGE$BGE_Suc)))

# Save BGE Data
write.csv(BGE, file="../data/BGE_data.txt", quote=FALSE, row.names=FALSE)
```

## Import Tree
```{r}
# Import Phylogeny (already rooted)
tree <- read.tree("../data/Phylogeny/HMWF.proteo.nwk.tre")

# Keep Rooted but Drop Outgroup Branch
tree <- drop.tip(tree, "Aquifex")
tree <- drop.tip(tree, c("HMWF015", "HMWF017", "HMWF032"))
tree$edge.length <- tree$edge.length + 10^-7
```

```{r}
# Carbon Respiration Traits {adephylo}
Resp <- BR[,c(11, 13, 15)]
Resp[is.na(Resp)] <- 0
Resp[Resp > 30] <- 30
colnames(Resp) <- c("Glucose", "Succinate", "Protocatechuate")
rownames(Resp) <- BR$Organism

# Reorder Resp
Resp <- as.matrix(Resp[match(tree$tip.label, row.names(Resp)), ])

# Carbon Production Traits
Prod <- BP[,c(11, 13, 15)]
Prod[is.na(Prod)] <- 0
Prod[Prod < 0] <- 0
Prod[Prod > 10] <- 10
colnames(Prod) <- c("Glucose", "Succinate", "Protocatechuate")
rownames(Prod) <- BP$Organism

# Reorder Prod
Prod <- as.matrix(Prod[match(tree$tip.label, row.names(Prod)), ])

# Carbon BGE Traits
BGE2 <- BGE[,2:4]
BGE2[is.na(BGE2)] <- 0
BGE2[BGE2 < 0] <- 0
BGE2[BGE2 > 0.9] <- 0
colnames(BGE2) <- c("Glu.", "Suc.", "Pro.")
rownames(BGE2) <- BGE$Organism

# Reorder BGE
BGE2 <- as.matrix(BGE2[match(tree$tip.label, row.names(BGE2)), ])
```

## Figure 1: BGE
```{r, results='hide'}
#png(filename="../figures/Figure1.png",
#    width = 1200, height = 1200, res = 96*2, bg = "white")
setEPS()
postscript("../figures/Figure_1.eps", width=6, height=6, bg = "white")

par(opar)

# Subset Tree
tree <- keep.tip(tree, row.names(BGE2))
tree$node.label <- c(1:tree$Nnode)

# Define Color Palette
mypalette <- colorRampPalette(brewer.pal(9, "YlOrRd"), bias = 2)

# Define Group
all.equal(tree$tip.label, row.names(BGE2))
Group <- rep(NA, dim(BGE2)[1])
for (i in 1:length(Group))
  if (!is.na(BGE2[i, 1])){
if (BGE2[i, 1] < 0.05){
  Group[i] <- "L"
} else {
  Group[i] <- "H"
}} else {}

# Make Plot
par(mar=c(0.5,0.5,0.5,1.5) + 0.1)
x <- phylo4d(tree, tip.data = BGE2[])

table.phylo4d2(x, treetype = "phylo", symbol = "colors", 
              show.node = FALSE, 
              cex.label = 1, scale = FALSE, use.edge.length = FALSE,
              edge.color = "black", edge.width = 3, box = FALSE,
              col=mypalette(25), pch = 22, cex.symbol = 2.5,
              ratio.tree = 0.7, cex.legend = 1.1, center = FALSE,
              show.tip.label = TRUE,
              tip.label = mapply(function(a,b)
      as.expression(bquote(.(a)[.(b)])), tipLabels(x), Group))

E <- phylobase::edges(x)
tips.ord <- E[, 2][!E[, 2] %in% E[, 1]]
#mtext(tipLabels(x), side = 4, las = 1, at = tips.ord, cex = 0.8)
#mtext(mapply(function(a,b)
#      as.expression(bquote(.(a)[.(b)])), tipLabels(x), Group), 
#      side = 4, las = 1, at = tips.ord, cex = 0.8)

text(15.5, 19.5, expression(alpha), cex=1.5)
text(15.5, 16.5, expression(beta), cex=1.5)
text(4.5, 10.5, expression(gamma), cex=1.5)
text(11, 13.5, "Xan.", cex=1)
text(11, 10.5, "Aero.", cex=1)
text(11, 6, "Pseu.", cex=1)
text(x = 4.7, y = 1.75, "BGE", cex = 1.5)

dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="BGE"}
img <- readPNG("../figures/Figure_1.png")
grid.raster(img)
```

## Organize Tree and Data
```{r}
# Hypothesis Testing
traits <- cbind(Resp, Prod, BGE2)

# Import Phylogeny (already rooted)
# tree1000 <- read.tree("../data/Phylogeny/RAxML_1000bootstrap.HMWF.tre")
R.tree <- read.tree("../data/Phylogeny/RAxML_bipartitions.RAxML_HMWF")

# Root Tree and Drop Extra Tipe
tree.rooted <- ape::root.phylo(R.tree, outgroup = "Aquifex")
tree.2 <- ape::drop.tip(tree.rooted, 
                       tip = c(setdiff(tree.rooted$tip.label, 
                       row.names(traits))))

# Reorder Traits
traits <- as.matrix(traits[match(tree.2$tip.label, row.names(traits)), ])
```

## Taxonomic and Resource Variance Partitioning
```{r}
BGE3 <- as.data.frame(BGE2)
BGE3$ID <- row.names(BGE3)
BGE3$Taxon <- c("Pseudo", "Pseudo", "Pseudo", "Pseudo", "Pseudo", 
                "Pseudo", "Pseudo", "Pseudo", "Aero", "Aero", 
                "Aero", "Xanto", "Xanto", "Xanto", "Beta", "Beta", 
                "Beta", "Alpha", "Alpha", "Alpha")
BGE3$TaxonP <- c("Gamma", "Gamma", "Gamma", "Gamma", "Gamma", 
                 "Gamma", "Gamma", "Gamma", "Gamma", "Gamma", 
                 "Gamma", "Gamma", "Gamma", "Gamma", "Beta", 
                 "Beta", "Beta", "Alpha", "Alpha", "Alpha")

# Reorganize Data
BGE.l <- melt(BGE3)

# Model 1: Isolate Identity 
fm1 <- lme(value ~ as.factor(ID), random=~1|variable, data = BGE.l)
summary(fm1)$AIC
r.squaredGLMM(fm1)

# Model 2: Taxonomic Order
fm2 <- lme(value ~ as.factor(Taxon), random=~1|variable, data = BGE.l)
summary(fm2)$AIC
r.squaredGLMM(fm2)

# Model 3: Resource Identity
fm3 <- lme(value ~ as.factor(variable), random=~1|ID, data = BGE.l)
summary(fm3)$AIC
r.squaredGLMM(fm3)
```

## Phylogenetic Patterns
```{r}
# Blomberg's K - BGE
## Glucose
phylosignal(traits[, 7], tree.2)
Kcalc(traits[, 7], tree.2)

## Succinate
phylosignal(traits[, 8], tree.2)
Kcalc(traits[, 8], tree.2)

## Protocatechuate
phylosignal(traits[, 9], tree.2)
Kcalc(traits[, 9], tree.2)

# Pagel's Lambda - BGE
## Glucose
phylosig(tree.2, traits[, 7], method = "lambda", test = T)

# Succinate
phylosig(tree.2, traits[, 8], method = "lambda", test = T)

# Protocatechuate
phylosig(tree.2, traits[, 9], method = "lambda", test = T)



# Generate Test Statistics for Comparing Phylogenetic Signal {geiger}
#fitContinuous(nj.rooted, nb, model = "lambda")
#fitContinuous(nj.lambda.0, nb, model = "lambda")

# First, Correct for Zero Branch-Lengths on Our Tree
#nj.rooted$edge.length <- nj.rooted$edge.length + 10^-7

# Calculate Phylogenetic Signal for Growth on All Phosphorus Resources
# First, Create a Blank Output Matrix
#p.phylosignal <- matrix(NA, 6, 18)
#colnames(p.phylosignal) <- colnames(p.growth.std)
#rownames(p.phylosignal) <- c("K", "PIC.var.obs", "PIC.var.mean",
#                             "PIC.var.P", "PIC.var.z", "PIC.P.BH")

# Use a For Loop to Calculate Blomberg's K for Each Resource
#for (i in 1:18){
#  x <- as.matrix(p.growth.std[ ,i, drop = FALSE])
#  out <- phylosignal(x, nj.rooted)
#  p.phylosignal[1:5, i] <- round(t(out), 3)
#}

# Use the BH Correction on P-values:
#p.phylosignal[6, ] <- round(p.adjust(p.phylosignal[4, ], method = "BH"), 3)


# Calcualate Phylogenetic Signal for Niche Breadth
#signal.nb <- phylosignal(nb, nj.rooted)

```

## BGE Distribution: Dip Test Glucose
```{r, results = "hide"}
png(filename="../figures/Density.png",
    width = 1200, height = 800, res = 96*2, bg = "white")
par(opar)

dip.test(na.omit(BGE2[,1]), simulate.p.value = T)

par(mar = c(4, 4, 1, 1) + 0.5)
plot(density(na.omit(BGE2[,1])), main = "Dip Test: Glucose",
     ylab = "", cex.lab = 1.25, 
     cex = 1.5, las = 1, lwd = 2, cex.axis = 1.25)
axis(1, lwd = 2, labels = F, las = 1, cex = 1.25)
axis(2, lwd = 2, labels = F, las = 1, cex = 1.25)
axis(1, lwd = 2, labels = F, tck = 0.02, las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, las = 1)
axis(3, lwd = 2, labels = F, tck = -0.02, las = 1)
axis(4, lwd = 2, labels = F, tck = -0.02, las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, las = 1)
mtext("Density", side = 2, line = 2.5, cex = 1.5)
box(lwd = 2)


dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
#```

#```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Kernel Density Plot"}
img <- readPNG("../figures/Density.png")
grid.raster(img)
```

## BGE Distribution: Dip Test Succinate
```{r, results = "hide"}
png(filename="../figures/Density2.png",
    width = 1200, height = 800, res = 96*2, bg = "white")
par(opar)

dip.test(na.omit(BGE2[,2]), simulate.p.value = T)

par(mar = c(4, 4, 1, 1) + 0.5)
plot(density(na.omit(BGE2[,2])), main = "Dip Test: Succinate",
     ylab = "", cex.lab = 1.25, 
     cex = 1.5, las = 1, lwd = 2, cex.axis = 1.25)
axis(1, lwd = 2, labels = F, las = 1, cex = 1.25)
axis(2, lwd = 2, labels = F, las = 1, cex = 1.25)
axis(1, lwd = 2, labels = F, tck = 0.02, las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, las = 1)
axis(3, lwd = 2, labels = F, tck = -0.02, las = 1)
axis(4, lwd = 2, labels = F, tck = -0.02, las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, las = 1)
mtext("Density", side = 2, line = 2.5, cex = 1.5)
box(lwd = 2)


dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
#```

#```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Kernel Density Plot"}
img <- readPNG("../figures/Density2.png")
grid.raster(img)
```

# Phenotypic Comparisions
Raw Data Last Updates
- umax: 8 Oct 2019 (check names and rerun, using old output for now)
- copy number: No updates needed
- ecoplate: 8 Oct 2019
- maple: 8 Oct 2019
```{r}
# Import Data
umax <- read.csv("../data/umax.txt", header=T) 
copies <- read.delim("../data/16SrRNA.txt")
ecoplate <- read.delim("../data/eco.data.txt") 
maple.raw <- read.delim("../data/Maple.MCR.txt", header = T, row.names = 1)
genomes <- read.delim("../data/Genomes.txt", header = T)

# Trim duplicates from umax
umax <- umax[-c(grep("^#", umax$isolate)), ]

BGE4 <- as.data.frame(BGE2[order(rownames(BGE2)), ])
BGE4$Organism <- rownames(BGE4)
colnames(BGE4) <- c("Glu", "Suc", "Pro", "Organism")

# Raw Data Calculations 
maple <- (maple.raw >= 80) * 1
pathways <- rowSums(maple)
pathways <- data.frame(Organism = names(pathways), pathways)
ecoplate.std <- ecoplate[,2:32] / (apply(ecoplate[,2:32], 1, sum))
ecoplate$levins <- levins(ecoplate.std)

ecoplate2 <- ecoplate[ecoplate$Strain %in% BGE4$Organism, ]
pathways2 <- pathways[pathways$Organism %in% BGE4$Organism, ]
genomes2 <- genomes[genomes$Strain %in% BGE4$Organism, ]
pathways3 <- merge(pathways2, genomes2, by.x = "Organism", by.y = "Strain")

# Build Dataframes
growth <- merge(umax, copies, by.x = "isolate", by.y = "Organism")
growth.bge <- merge(BGE4, growth, by.x = "Organism", by.y = "isolate")

res.niche <- merge(pathways3, ecoplate2[,c(1,33,34)], 
               by.x = "Organism", by.y = "Strain", all.x = T)
niche.bge <- merge(BGE4, res.niche)

# Define Groups
growth.bge$group <- NA
for (i in 1:dim(growth.bge)[1])
  if (!is.na(growth.bge$Glu[i])){
if (growth.bge$Glu[i] < 0.05){
  growth.bge$group[i] <- "A"
} else {
  growth.bge$group[i] <- "B"
}} else {}

aggregate(growth.bge$Glu, list(group = growth.bge$group), 
          function(x) c(avg = mean(x), se = se(x)))

niche.bge$group <- NA
for (i in 1:dim(niche.bge)[1])
  if (!is.na(niche.bge$Glu[i])){
if (niche.bge$Glu[i] < 0.05){
  niche.bge$group[i] <- "A"
} else {
  niche.bge$group[i] <- "B"
}} else {}

# Physiological Tradeoffs
## Growth Rate
data <- growth.bge
mod1 <- lm(Glu ~ umax * group, data = data)
summary(mod1)

# High BGE
mod2 <- lm(Glu ~ umax, data = data[data$group == "B", ])
summary(mod2)

# Low BGE
mod3 <- lm(Glu ~ umax, data = data[data$group == "A", ])
summary(mod3)

## Niche Breadth
data <- niche.bge
mod1 <- lm(Glu ~ levins * group, data = data)
summary(mod1)

# High BGE
mod2 <- lm(Glu ~ levins, data = data[data$group == "B", ])
summary(mod2)

# Low BGE
mod3 <- lm(Glu ~ levins, data = data[data$group == "A", ])
summary(mod3)

```

## Figure 2: Species Traits
```{r, results='hide'}
#png(filename="../figures/Figure_2.png",
#    width = 1600, height = 800, res = 96*2, bg = "white")
setEPS()
postscript("../figures/Figure_2.eps", width=9, height=4.5, bg = "white")


par(opar)

layout(matrix(1:2, nrow = 1))

par(mar = c(3,0.5,0.5,0.5) + 0.1, oma = c(2, 5, 1, 1))

# Maximum Growth Rate
groupA <- lm(Glu ~ umax, data = growth.bge[growth.bge$group == "A", ])
groupB <- lm(Glu ~ umax, data = growth.bge[growth.bge$group == "B", ])
pred.frameA <- data.frame(umax = seq(0.04, 0.12, by = 0.01))
pred.frameB <- data.frame(umax = seq(0.05, 0.20, by = 0.01))

plot(Glu ~ umax, growth.bge, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(0.04, 0.20), ylim = c(0,0.35), las = 1,
     pch = 22, bg = "gray", lwd = 2, cex = 1.5)
add.hull(groupA, pred.frameA)
matlines(pred.frameA, predict(groupA, interval = "c", newdata=pred.frameA),
         lty=c(2,3,3), lwd=c(4,2,2), col="black")
add.hull(groupB, pred.frameB)
matlines(pred.frameB, predict(groupB, interval = "c", newdata=pred.frameB),
         lty=c(2,3,3), lwd=c(4,2,2), col="black")
points(Glu ~ umax, growth.bge[growth.bge$group == "A", ],
       pch = 22, bg = "gray", lwd = 2, cex = 1.5)
points(Glu ~ umax, growth.bge[growth.bge$group == "B", ],
       pch = 24, bg = "gray", lwd = 2, cex = 1.5)
axis(1, lwd = 2, labels = T, at = c(seq(0, 0.2, 0.05)), las = 1)
axis(2, lwd = 2, labels = T, at = c(0, 0.1, 0.2, 0.3), las = 1)
axis(1, lwd = 2, labels = F, tck = 0.02, at = c(seq(0, 0.2, 0.05)), las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, at = c(0, 0.1, 0.2, 0.3), las = 1)
axis(3, lwd = 2, labels = F, tck = -0.02, at = c(seq(0, 0.2, 0.05)), las = 1)
axis(4, lwd = 2, labels = F, tck = -0.02, at = c(0, 0.1, 0.2, 0.3), las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, at = c(seq(0, 0.2, 0.05)), las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, at = c(0, 0.1, 0.2, 0.3), las = 1)
mtext(expression(paste("Maximum Growth Rate (hr"^-1,")")), side = 1, line = 3, cex = 1.5)
#mtext("Oligotroph-Copiotroph", side = 3, line = 1)
mtext("Bacterial Growth Efficiency", side = 2, line = 2.75, cex = 1.5)
box(lwd = 2)

# Add Legend
legend("topright", c("High BGE Group", "Low BGE Group"), pch = c(24, 22), pt.bg = "gray", pt.cex = 1, pt.lwd = 2, bty = "n", cex = 0.8)

# Niche Breadth
groupA <- lm(Glu ~ levins, data = niche.bge[niche.bge$group == "A", ])
groupB <- lm(Glu ~ levins, data = niche.bge[niche.bge$group == "B", ])
pred.frameA <- data.frame(levins = seq(0.04, 0.45, by = 0.01))
pred.frameB <- data.frame(levins = seq(0.04, 0.5, by = 0.01))

plot(Glu ~ levins, niche.bge, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(0.04, 0.5), ylim = c(0, 0.35), las = 1,
     pch = 22, bg = "gray", lwd = 2, cex = 1.5)
add.hull(groupA, pred.frameA)
matlines(pred.frameA, predict(groupA, interval = "c", newdata=pred.frameA),
         lty=c(2,3,3), lwd=c(4,2,2), col="black")
add.hull(groupB, pred.frameB)
matlines(pred.frameB, predict(groupB, interval = "c", newdata=pred.frameB),
         lty=c(2,3,3), lwd=c(4,2,2), col="black")
points(Glu ~ levins, niche.bge[niche.bge$group == "A", ],
       pch = 22, bg = "gray", lwd = 2, cex = 1.5)
points(Glu ~ levins, niche.bge[niche.bge$group == "B", ],
       pch = 24, bg = "gray", lwd = 2, cex = 1.5)
axis(1, lwd = 2, labels = T, at = c(seq(0, 0.5, 0.1)), las = 1)
axis(2, lwd = 2, labels = F, at = c(0, 0.1, 0.2, 0.3), las = 1)
axis(1, lwd = 2, labels = F, tck = 0.02, at = c(seq(0, 0.5, 0.1)), las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, at = c(0, 0.1, 0.2, 0.3), las = 1)
axis(3, lwd = 2, labels = F, tck = -0.02, at = c(seq(0, 0.5, 0.1)), las = 1)
axis(4, lwd = 2, labels = F, tck = -0.02, at = c(0, 0.1, 0.2, 0.3), las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, at = c(seq(0, 0.5, 0.1)), las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, at = c(0, 0.1, 0.2, 0.3), las = 1)
mtext("Levins Index", side = 1, line = 3, cex = 1.5)
#mtext("Generalist-Specialist", side = 3, line = 1)
box(lwd = 2)

dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="Traits"}
img <- readPNG("../figures/Figure_2.png")
grid.raster(img)
```



# Genomic Comparisions
## Number of Pathways
```{r}
# Number of Pathways (Group A = low BGE; Group B = high BGE)
data <- niche.bge
mod.p <- lm(Glu ~ pathways * group, data = data)
summary(mod.p)

mod.b <- lm(Glu ~ bp * group, data = data)
summary(mod.b)

mod.g <- lm(Glu ~ genes * group, data = data)
summary(mod.g)

# Low BGE Group
mod1 <- lm(Glu ~ pathways, data = data[data$group == "A", ])
summary(mod1)

# High BGE Group
mod2 <- lm(Glu ~ pathways, data = data[data$group == "B", ])
summary(mod2)

t.test(data$pathways ~ data$group)
aggregate(data$pathways, list(group = data$group), 
          function(x) c(avg = mean(x), se = se(x)))
```

## Composition of Pathways
```{r}
# Permanova
str(maple)
maple2 <- maple[which(rownames(maple) %in% BGE4$Organism), ]
maple2 <- maple2[, which(colSums(maple2) > 0)]

all.equal(rownames(maple2), data$Organism)

t.test(rowSums(maple2) ~ data$group)

adonis(maple2 ~ as.factor(data$group), method = "bray")
maple.dist <- vegdist(maple2, method = "bray", binary = T)
maple.beta <- betadisper(maple.dist, as.factor(data$group))
anova(maple.beta)
maple.beta
maple.beta

# No evidence of differences based on permanova
# using betadispersion, there are differences, but tey seem to be from two of the isolates. Likely this has nothing to do with BGE. 


all.equal(rownames(maple2), data$Organism)
adonis(maple2 ~ BGE4$Glu * as.factor(data$group), method = "bray")
# Not significant

adonis(maple2 ~ as.factor(data$group), method = "bray")

inds <- indval(maple2, as.factor(data$group)) 
summary(inds)

# Just in the high-BGE group
mapleB <- maple2[data$group == "B", ]
BGE_glu_B <- BGE4$Glu[data$group == "B"]

# Calculate Bray-Curtis Distances for Genomes
mapleB.bray <- vegdist(mapleB, "bray")

# BGE dbRDA 
BGE.dbRDA.B <- capscale(mapleB.bray ~ BGE_glu_B, comm = mapleB, 
                      add = T, na.action = na.exclude)
summary(BGE.dbRDA.B)
RsquareAdj(BGE.dbRDA.B)
anova(BGE.dbRDA.B)

# Just in group low BGE group
mapleA <- maple2[data$group == "A", ]
BGE_glu_A <- BGE4$Glu[data$group == "A"]

# Calculate Bray-Curtis Distances for Genomes
mapleA.bray <- vegdist(mapleA, "bray")

# BGE dbRDA 
BGE.dbRDA.A <- capscale(mapleA.bray ~ BGE_glu_A, comm = mapleA, 
                      add = T, na.action = na.exclude)
summary(BGE.dbRDA.A)
RsquareAdj(BGE.dbRDA.A)
anova(BGE.dbRDA.A)

# Pathways Scores for high BGE
pathway.scores <- data.frame(matrix(NA, dim(mapleB)[2], 2))
colnames(pathway.scores) <- c("pathway", "cor")
pathway.scores[,1] <- colnames(mapleB)
for (i in 1:dim(mapleB)[2]){
  pathway.scores[i,2] <- cor(BGE_glu_B,mapleB[,i], method = "spearman")
  #pathway.scores[i,2] <- cor(BGE.dbRDA.B$CCA$wa,mapleB[,i], method = "spearman")
}

sig.path <- pathway.scores[which(abs(pathway.scores$cor) >= 0.70), ]
sig.path

```




# Physiological Constraints
```{r}
# Organize Data
BGE4 
BGE4$Group <- NA
for (i in 1:dim(BGE4)[1])
  if (!is.na(BGE4$Glu[i])){
if (BGE4$Glu[i] < 0.05){
  BGE4$Group[i] <- "A"
} else {
  BGE4$Group[i] <- "B"
}} else {}

all.equal(as.character(BR$Organism), rownames(BGE4))

Phys <- data.frame(Organism = BR$Organism,
                   RespGlu = BR$Glu_mean, RespSuc = BR$Suc_mean, 
                   RespPro = BR$Pro_mean, 
                   ProdGlu = BP$Glu_mean, ProdSuc = BP$Suc_mean, 
                   ProdPro = BP$Pro_mean, 
                   BGEGlu = BGE4$Glu, BGESuc = BGE4$Suc,
                   BGEPro = BGE4$Pro,
                   Group = BGE4$Group)

Phys.full <- data.frame(Organism = rep(BGE4$Organism, 3), 
                        Group = rep(BGE4$Group, 3),
                        Resource = c(rep("Glu", dim(BR)[1]),
                                     rep("Suc", dim(BR)[1]),
                                     rep("Pro", dim(BR)[1])),
                        Resp = c(BR$Glu_mean, BR$Suc_mean, BR$Pro_mean),
                        Prod = c(BP$Glu_mean, BP$Suc_mean, BP$Pro_mean),
                        BGE = c(BGE4$Glu, BGE4$Suc, BGE4$Pro))

# Full Model
mod1 <- glm(log10(Resp) ~ (log10(Prod) + Group + Resource)^2, data = Phys.full)
summary(mod1)

mod2 <- glm(log1p(Resp) ~ log1p(Prod) * Resource, data = Phys.full)
summary(mod2)

mod3 <- glm(log1p(Resp) ~ log1p(Prod) * Group, data = Phys.full)
summary(mod3)

mod4 <- glm(log1p(Resp) ~ log1p(Prod), data = Phys.full)
summary(mod4)

# Independent Models
modA <- glm(log1p(Resp) ~ log1p(Prod), 
           data = Phys.full[Phys.full$Group == "A", ])
summary(modA)
modB <- glm(log1p(Resp) ~ log1p(Prod), 
           data = Phys.full[Phys.full$Group == "B", ])
summary(modB)

Phys2 <- Phys
Phys2$RespGlu <- log10(Phys2$RespGlu)
Phys2$ProdGlu <- log10(Phys2$ProdGlu)
Phys2$RespSuc <- log10(Phys2$RespSuc)
Phys2$ProdSuc <- log10(Phys2$ProdSuc)
Phys2$RespPro <- log10(Phys2$RespPro)
Phys2$ProdPro <- log10(Phys2$ProdPro)
for (i in 2:7){
  Phys2[which(Phys2[, i] < -5), i] = 0 
}


# Maximum Growth Rate
glu.model <- lm(ProdGlu ~ RespGlu * Group, data = Phys2)
suc.model <- lm(ProdSuc ~ RespSuc * Group, data = Phys2)
pro.model <- lm(ProdPro ~ RespPro * Group, data = Phys2)


# Compare Models
PhysGlu <- Phys2[, c(2, 5, 11)]
colnames(PhysGlu) <- c("Resp", "Prod", "Group")
PhysSuc <- Phys2[, c(3, 6, 11)]
colnames(PhysSuc) <- c("Resp", "Prod", "Group")

glu.model <- lm(Prod ~ Resp * Group, data = PhysGlu)
suc.model <- lm(Prod ~ Resp * Group, data = PhysSuc)

test <- anova(glu.model, suc.model, test = "F")

PhysGlu <- Phys2[, c(1, 2, 5, 8, 11)]
colnames(PhysGlu) <- c("Organism", "Resp", "Prod", "BGE", "Group")
PhysGlu$Res <- rep("Glu", dim(PhysGlu)[1])
PhysSuc <- Phys2[, c(1, 3, 6, 9, 11)]
colnames(PhysSuc) <- c("Organism", "Resp", "Prod", "BGE", "Group")
PhysSuc$Res <- rep("Suc", dim(PhysSuc)[1])
PhysPro <- Phys2[, c(1, 4, 7, 10, 11)]
colnames(PhysPro) <- c("Organism", "Resp", "Prod", "BGE", "Group")
PhysPro$Res <- rep("Pro", dim(PhysPro)[1])
Phys.l <- rbind(PhysGlu, PhysSuc, PhysPro)
Phys.l2 <- na.omit(Phys.l)

Phys.l.b <- Phys.l2
Phys.l.b$Resp <- 10*exp(Phys.l.b$Resp)
Phys.l.b$Prod <- 10*exp(Phys.l.b$Prod)
Phys.l.b$BGE <- 10*exp(Phys.l.b$BGE)

mod1 <- lm(Prod ~ (Resp + Res + Group)^2, data = Phys.l2)
summary(mod1)

ttest <- function(reg, coefnum, val){
  co <- coef(summary(reg))
  tstat <- (co[coefnum,1]-val)/co[coefnum,2]
  pstat <- 2 * pt(abs(tstat), reg$df.residual, lower.tail = FALSE)
  return(list = c(tstat, reg$df.residual, pstat))
}

ttest(mod1, 2, 1)



```

## Figure 3
```{r, results = "hide"}
#png(filename="../figures/Figure_3.png",
#    width = 1200, height = 1200, res = 96*2, bg = "white")
setEPS()
postscript("../figures/Figure_3.eps", width=6, height=6, bg = "white")

par(opar)
par(mar = c(6,6,1,1) + 0.5)

Phys.l2$Prod[which(Phys.l2$Prod < -2.8)] <- NA

mod1 <- lm(Prod ~ (Resp + Group + Res)^2, data = Phys.l2)
summary(mod1)
ttest(mod1, 2, 1)

groupA <- lm(Prod ~ Resp, data = Phys.l2[Phys.l2$Group == "A" & Phys.l2$Res == "Glu", ])
groupB <- lm(Prod ~ Resp, data = Phys.l2[Phys.l2$Group == "B" & Phys.l2$Res == "Glu", ])
pred.frameA <- data.frame(Resp = seq(0.65, 3.2, by = 0.01))
pred.frameB <- data.frame(Resp = seq(0.38, 2.68, by = 0.01))

plot(Prod ~ Resp, Phys.l2, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(0.35,3.3), ylim = c(-3, 2.4), las = 1,
     pch = 22, bg = "gray", lwd = 2, cex = 1.5)


add.hull(groupA, pred.frameA)
matlines(pred.frameA, predict(groupA, interval = "c", newdata=pred.frameA),
        lty=c(2,3,3), lwd=c(4,2,2), col="black") 
add.hull(groupB, pred.frameB)
matlines(pred.frameB, predict(groupB, interval = "c", newdata=pred.frameB),
         lty=c(2,3,3), lwd=c(4,2,2), col="black")  


points(Prod ~ Resp, Phys.l2[Phys.l2$Group == "A" & Phys.l2$Res == "Glu", ],
       pch = 22, bg = "gray90", lwd = 2, 
       cex = 10^c(Phys.l2$BGE[Phys.l2$Group == "A" & Phys.l2$Res == "Glu"]) * 1.2) 
points(Prod ~ Resp, Phys.l2[Phys.l2$Group == "B" & Phys.l2$Res == "Glu", ],
       pch = 24, bg = "gray90", lwd = 2, 
       cex = 10^c(Phys.l2$BGE[Phys.l2$Group == "B" & Phys.l2$Res == "Glu"]) * 1.2) 

points(Prod ~ Resp, Phys.l2[Phys.l2$Group == "A" & Phys.l2$Res == "Suc", ],
       pch = 22, bg = "gray60", lwd = 2, 
       cex = 10^c(Phys.l2$BGE[Phys.l2$Group == "A" & Phys.l2$Res == "Suc"]) * 1.2) 
points(Prod ~ Resp, Phys.l2[Phys.l2$Group == "B" & Phys.l2$Res == "Suc", ],
       pch = 24, bg = "gray60", lwd = 2, 
       cex = 10^c(Phys.l2$BGE[Phys.l2$Group == "B" & Phys.l2$Res == "Suc"]) * 1.2) 

points(Prod ~ Resp, Phys.l2[Phys.l2$Group == "A" & Phys.l2$Res == "Pro", ],
       pch = 22, bg = "gray20", lwd = 2, 
       cex = 10^c(Phys.l2$BGE[Phys.l2$Group == "A" & Phys.l2$Res == "Pro"]) * 1.2) 
points(Prod ~ Resp, Phys.l2[Phys.l2$Group == "B" & Phys.l2$Res == "Pro", ],
       pch = 24, bg = "gray20", lwd = 2, 
       cex = 10^c(Phys.l2$BGE[Phys.l2$Group == "B" & Phys.l2$Res == "Pro"]) * 1.2)

#text(-3.8, 0.52, paste("*", round(groupA1$coefficients[1], 2), sep = ""), cex = 1.2)
#text(-2, -1, paste("*", round(groupB1$coefficients[1], 2), sep = ""), , cex = 1.2) 

axis(1, lwd = 2, labels = c(2, 10, 100, 1000), 
     at = c(log10(c(2,10,100,1000))), las = 1)
axis(1, lwd = 2, labels = F, tck = 0.02, 
     at = c(log10(c(2,10,100,1000))), las = 1)

axis(2, lwd = 2, labels = c(0.001, 0.01, 0.1, 1, 10, 100), 
     at = c(log10(c(0.001, 0.01, 0.1, 1, 10, 100))), las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, 
    at = c(log10(c(0.001, 0.01, 0.1, 1, 10, 100))), las = 1)

axis(3, lwd = 2, labels = F, tck = -0.02, 
     at = c(log10(c(2,10,100,1000))), las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, 
     at = c(log10(c(2,10,100,1000))), las = 1)

axis(4, lwd = 2, labels = F, tck = -0.02, 
     at = c(log10(c(0.001, 0.01, 0.1, 1, 10, 100))), las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, 
     at = c(log10(c(0.001, 0.01, 0.1, 1, 10, 100))), las = 1)

mtext("Production Rate", side = 2, line = 4.5, cex = 1.5)
mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
                       side = 2, line = 3, cex = 1)
mtext("Respiration Rate", side = 1, line = 2.5, cex = 1.5)
mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
                       side = 1, line = 4, cex = 1)
box(lwd = 2)

# Add Legend
legend("bottomright", c("High BGE Group", "Low BGE Group"), 
       pch = c(24, 22), pt.bg = "gray", pt.cex = 1, pt.lwd = 2, 
       bty = "n", cex = 0.8, inset = 0.05)
legend("topleft", c("Glucose", "Succinate", "Protocatechuate"), 
       pch = 22, pt.bg = c("gray90", "gray60", "gray20"), 
       pt.cex = 1, pt.lwd = 2, 
       bty = "n", cex = 0.8, inset = 0.05)

text(x = 2.7, y = 2.12, "high", cex = 1.25, srt = 29)
text(x = 3.18, y = 0.965, "low", cex = 1.25, srt = 31)


dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
```

```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="BGE"}
img <- readPNG("../figures/Figure_3.png")
grid.raster(img)
```

## Extra Figure
```{r, results='hide'}
png(filename="../figures/AllResources.png",
    width = 800, height = 1800, res = 96*2, bg = "white")
par(opar)

layout(matrix(1:3, nrow = 3))

par(mar = c(0.25,6,0.25,1) + 0.5, oma = c(7, 1, 1, 1) + 0.5)

# Glucose
groupA <- lm(ProdGlu ~ RespGlu, data = Phys2[Phys2$Group == "A", ])
groupB <- lm(ProdGlu ~ RespGlu, data = Phys2[Phys2$Group == "B", ])
pred.frameA <- data.frame(RespGlu = seq(0.5, 3.2, by = 0.1))
pred.frameB <- data.frame(RespGlu = seq(0.3, 1.8, by = 0.1))

plot(ProdGlu ~ RespGlu, Phys2, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(0.3, 3.2), ylim = c(-3,2.2), las = 1,
     pch = 22, bg = "gray", lwd = 2, cex = 1.5)
add.hull(groupA, pred.frameA)
matlines(pred.frameA, predict(groupA, interval = "c", newdata=pred.frameA),
         lty=c(2,3,3), lwd=c(4,2,2), col="black") 
add.hull(groupB, pred.frameB)
matlines(pred.frameB, predict(groupB, interval = "c", newdata=pred.frameB),
         lty=c(2,3,3), lwd=c(4,2,2), col="black")  
points(ProdGlu ~ RespGlu, Phys2[Phys2$Group == "A", ],
       pch = 22, bg = "gray", lwd = 2, cex = 1.2 + Phys2$BGEGlu[Phys2$Group == "A"] * 5) 
points(ProdGlu ~ RespGlu, Phys2[Phys2$Group == "B", ],
       pch = 24, bg = "gray", lwd = 2, cex = 1.2 + Phys2$BGEGlu[Phys2$Group == "B"] * 5)  
axis(1, lwd = 2, labels = F, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(2, lwd = 2, labels = c(0.001, 0.01, 0.1, 1, 10, 100), at = c(seq(-3, 2, 1)), las = 1)
axis(1, lwd = 2, labels = F, tck = 0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, at = c(seq(-3, 2, 1)), las = 1)
axis(3, lwd = 2, labels = F, tck = -0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(4, lwd = 2, labels = F, tck = -0.02, at = c(seq(-3, 2, 1)), las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, at = c(seq(-3, 2, 1)), las = 1)
#mtext("Respiration Rate", side = 1, line = 2.5, cex = 1.5)
#mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
#                       side = 1, line = 4, cex = 1)
#mtext("Production Rate", side = 2, line = 4.5, cex = 1.5)
#mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
#                       side = 2, line = 3, cex = 1)
box(lwd = 2)

# Add Legend
#legend("bottomright", c("High BGE Group", "Low BGE Group"), 
 #      pch = c(24, 22), pt.bg = "gray", pt.cex = 1, pt.lwd = 2, 
 #      bty = "n", cex = 0.8)
legend("topleft", "Glucose", bty = "n")

# Succinate
groupA <- lm(ProdSuc ~ RespSuc, data = Phys2[Phys2$Group == "A", ])
groupB <- lm(ProdSuc ~ RespSuc, data = Phys2[Phys2$Group == "B", ])
pred.frameA <- data.frame(RespSuc = seq(0.7, 2.5, by = 0.01))
pred.frameB <- data.frame(RespSuc = seq(0.89, 2.77, by = 0.01))

plot(ProdSuc ~ RespSuc, Phys2, axes = F, type = "n",
     xlab = "", ylab = "",
     xlim = c(0.3, 3.2), ylim = c(-3.2, 2.2), las = 1,
     pch = 22, bg = "gray", lwd = 2, cex = 1.5)
add.hull(groupA, pred.frameA)
matlines(pred.frameA, predict(groupA, interval = "c", newdata=pred.frameA),
         lty=c(2,3,3), lwd=c(4,2,2), col="black") 
add.hull(groupB, pred.frameB)
matlines(pred.frameB, predict(groupB, interval = "c", newdata=pred.frameB),
         lty=c(2,3,3), lwd=c(4,2,2), col="black")  
points(ProdSuc ~ RespSuc, Phys2[Phys2$Group == "A", ],
       pch = 22, bg = "gray", lwd = 2, cex = 1.2 + Phys2$BGESuc[Phys2$Group == "A"] * 5) 
points(ProdSuc ~ RespSuc, Phys2[Phys2$Group == "B", ],
       pch = 24, bg = "gray", lwd = 2, cex = 1.2 + Phys2$BGESuc[Phys2$Group == "B"] * 5)  
axis(1, lwd = 2, labels = F, at = c(0.3, seq(1, 4, 1)), las = 1)
axis(2, lwd = 2, labels = c(0.001, 0.01, 0.1, 1, 10, 100), at = c(seq(-3, 2, 1)), las = 1)
axis(1, lwd = 2, labels = F, tck = 0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, at = c(seq(-3, 2, 1)), las = 1)
axis(3, lwd = 2, labels = F, tck = -0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(4, lwd = 2, labels = F, tck = -0.02, at = c(seq(-3, 2, 1)), las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, at = c(seq(-3, 2, 1)), las = 1)
#mtext("Respiration Rate", side = 1, line = 2.5, cex = 1.5)
#mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
#                       side = 1, line = 4, cex = 1)
mtext("Production Rate", side = 2, line = 5, cex = 1.5)
mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
                       side = 2, line = 3.5, cex = 1)
box(lwd = 2)

# Add Legend
#legend("bottomright", c("High BGE Group", "Low BGE Group"), 
#       pch = c(24, 22), pt.bg = "gray", pt.cex = 1, pt.lwd = 2, 
#       bty = "n", cex = 0.8, inset = 0.02)
legend("topleft", "Succinate", bty = "n")

# Protocatechuate
pro.model<- lm(ProdPro ~ RespPro, data = Phys2)
groupA <- lm(ProdPro ~ RespPro, data = Phys2[Phys2$Group == "A", ])
groupB <- lm(ProdPro ~ RespPro, data = Phys2[Phys2$Group == "B", ])
pred.frameA <- data.frame(RespPro = seq(1, 3.2, by = 0.1))
pred.frameB <- data.frame(RespPro = seq(0.89, 2.77, by = 0.01))

plot(ProdPro ~ RespPro, Phys2, axes = F, type = "n", 
     xlab = "", ylab = "",
     xlim = c(0.3, 3.2), ylim = c(-2.2, 2.2), las = 1,
     pch = 22, bg = "gray", lwd = 2, cex = 1.5) 
add.hull(pro.model, pred.frameA)
matlines(pred.frameA, predict(pro.model, interval = "c", newdata=pred.frameA),
         lty=c(2,3,3), lwd=c(4,2,2), col="black") 
#add.hull(groupB, pred.frameB)
#matlines(pred.frameB, predict(groupB, interval = "c", newdata=pred.frameB),
#         lty=c(2,3,3), lwd=c(4,2,2), col="black")  
points(ProdPro ~ RespPro, Phys2[Phys2$Group == "A", ],
       pch = 22, bg = "gray", lwd = 2, cex = 1.2 + Phys2$BGEPro[Phys2$Group == "A"] * 5) 
points(ProdPro ~ RespPro, Phys2[Phys2$Group == "B", ],
       pch = 24, bg = "gray", lwd = 2, cex = 1.2 + Phys2$BGEPro[Phys2$Group == "B"] * 5)  
axis(1, lwd = 2, labels = c(2, 10, 100, 1000, 10000), at = c(0.3, seq(1, 4, 1)), las = 1)
axis(2, lwd = 2, labels = c(0.001, 0.01, 0.1, 1, 10, 100), at = c(seq(-3, 2, 1)), las = 1)
axis(1, lwd = 2, labels = F, tck = 0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(2, lwd = 2, labels = F, tck = 0.02, at = c(seq(-3, 2, 1)), las = 1)
axis(3, lwd = 2, labels = F, tck = -0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(4, lwd = 2, labels = F, tck = -0.02, at = c(seq(-3, 2, 1)), las = 1)
axis(3, lwd = 2, labels = F, tck = 0.02, at = c(0.3, seq(1, 3, 1)), las = 1)
axis(4, lwd = 2, labels = F, tck = 0.02, at = c(seq(-3, 2, 1)), las = 1)
mtext("Respiration Rate", side = 1, line = 3.5, cex = 1.5)
mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
                       side = 1, line = 5, cex = 1)
#mtext("Production Rate", side = 2, line = 4.5, cex = 1.5)
#mtext(expression(paste("(pmole C cell"^-1, "hr"^-1, ")")), 
#                       side = 2, line = 3, cex = 1)
box(lwd = 2)

# Add Legend
legend("bottomright", c("High BGE Group", "Low BGE Group"), pch = c(24, 22), pt.bg = "gray", pt.cex = 1, pt.lwd = 2, bty = "n", cex = 0.8, inset = 0.02)
legend("topleft", "Protocatechuate", bty = "n")


dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
# ```

# ```{r fig.width=4, fig.height=4,echo=FALSE,fig.cap="BGE"}
img <- readPNG("../figures/AllResources.png")
grid.raster(img)
```


# Supplemental 
## Supp Figure 1: Phylogenetic Tree
Do I want to make a really nice tree or just use the species tree in the next figure
```{r, results = "hide"}
# Import Phylogenetic Tree
tree.raw <- read.tree("../data/Phylogeny/RAxML_bipartitions.RAxML_HMWF")
tree.rooted <- ape::root(tree.raw, "Aquifex", resolve.root = T)

tree.rooted$node.label

plot.phylo(tree.rooted, type = "phylogram", 
           direction = "right", show.tip.label=TRUE,
           use.edge.length = FALSE, cex = 0.6, label.offset = 1, 
           main = "Seq w/o Neighbors w/o Rooting")
add.scale.bar(cex = 0.7)
nodelabels(tree.rooted$node.label, font = 2, bg = "white", 
           frame = "none", cex = 0.4, adj = c(1.2, 1.2), col = "red")
```


